maths/points_are_collinear_3d.py

```diff
@@ -0,0 +1,136 @@
+"""
+Check if three points are collinear in 3D.
+
+In short, the idea is that we are able to create a triangle using three points,
+and the area of that triangle can determine if the three points are collinear or not.
+
+
+First we well create tow vectors with the same initial point from the three points,
+then we will calcolate the cross product of them.
+
+The length of the cross vector is numerically equal to the area of a parallelogram.
+
+Finally the area of the triangle is equal to the half of the area of the parallelogram.
+
+Since we are only differentiating between zero and anything else,
+we can get rid of the square root when calculating the length of the vector,
+and also the division by two at the end.
+
+From a second perspective, if the two vectors are parallel and overlapping,
+we can't get a nonzero perpendicular vector,
+since there will be an infinite number of orthogonal vectors.
+
+To simplify the solution we will not calculate the length,
+but we will decide directly from the vector whether it is equal to (0, 0, 0) or not.
+
+
+Read More:
+    https://math.stackexchange.com/a/1951650
+"""
+
+Vector = tuple[float, float, float]
+Point = tuple[float, float, float]
```

Comment:
This is a type-aliase.

Read More:
https://docs.python.org/3/library/typing.html#type-aliases

Reply:
Would `Vector3d` and `Point3d` be more self-documenting names?  Points are usually just x, y.
Would it be a good idea to create a `Triangle3d` instead of passing around three points?